### PR TITLE
Add stylelint

### LINKED ISSUE #1253: Add Sass linter to project
Lint the project's Sass 
https://stylelint.io/

### PR DESCRIPTION
**Pull request by maya**
## Related Issue or Background Info

Fixes #1253

### Plan
1. Merge this PR
2. (In subsequent work) investigate the `no-descending-specificity`, `at-rule-no-unknown` rules, see if we can fix them, and remove those ignore rules from the config

### CODE REVIEW

File: frontend/src/scss/components/card.scss [hunk 1]
```diff
@@ -1,5 +1,6 @@
 @mixin card($border-radius, $padding: 0) {
   @include u-padding($padding);
+
```
Comment (maya) [inline frontend/src/scss/components/card.scss:3]: Ideally, we can suppress adding a space after the include when the include is in the beginning of the declaration. Will need to look into this.

**Comment by maya** [comment]
The issues we had were all related to `no-descending-specificity` (suppressed now):
```
src/scss/PrimeStyles.scss
 135:1  ✖  Expected selector ".prime-test-name" to come before selector ".prime-queue-item .prime-test-name"       no-descending-specificity
 135:1  ✖  Expected selector ".prime-test-name" to come before selector ".prime-queue-item__ready                  no-descending-specificity
           .prime-test-name"
 307:1  ✖  Expected selector ".modal__close-button" to come before selector ".modal--basic .modal__close-button"   no-descending-specificity
 435:7  ✖  Expected selector ".usa-nav__primary .nav__primary-item-icon > a" to come before selector               no-descending-specificity
           ".usa-nav__primary .usa-nav__primary-item:first-child a"
 564:1  ✖  Expected selector ".usa-select" to come before selector ".prime-facility-select .usa-select"            no-descending-specificity
 599:1  ✖  Expected selector ".ReactModal__Content" to come before selector ".modal--basic .ReactModal__Content"   no-descending-specificity
 606:3  ✖  Expected selector ".ReactModal__Content .prime-formgroup" to come before selector                       no-descending-specificity
           ".ReactModal__Content .message--required + .prime-formgroup"
 642:7  ✖  Expected selector ".ReactModal__Content .symptom-checkboxes > .usa-checkbox" to come before selector    no-descending-specificity
           ".ReactModal__Content .symptom-checkboxes .usa-form-group .usa-checkbox:first-child"
 643:9  ✖  Expected selector ".ReactModal__Content .symptom-checkboxes > .usa-checkbox .usa-checkbox__label" to    no-descending-specificity
           come before selector ".ReactModal__Content .symptom-checkboxes .usa-form-group
           .usa-checkbox:first-child .usa-checkbox__label"
 648:7  ✖  Expected selector ".ReactModal__Content .symptom-checkboxes .usa-checkbox__label" to come before        no-descending-specificity
           selector ".ReactModal__Content .symptom-checkboxes .usa-form-group .usa-checkbox:first-child
           .usa-checkbox__label"
 648:7  ✖  Expected selector ".ReactModal__Content .symptom-checkboxes .usa-checkbox__label" to come before        no-descending-specificity
           selector ".ReactModal__Content .symptom-checkboxes > .usa-checkbox .usa-checkbox__label"
 690:1  ✖  Expected selector ".usa-step-indicator__segment" to come before selector                                no-descending-specificity
           ".usa-step-indicator--counters-sm .usa-step-indicator__segment"
 690:1  ✖  Expected selector ".usa-step-indicator__segment" to come before selector                                no-descending-specificity
           ".usa-step-indicator--counters-sm .usa-step-indicator__segment:last-child"
 764:3  ✖  Expected selector ".tos-content h2" to come before selector ".prime-container .usa-card__header h2"     no-descending-specificity

src/app/patients/EditPatient.scss
 1:1  ✖  Unexpected empty source   no-empty-source

src/app/Settings/Users/ManageUsers.scss
 30:1  ✖  Expected selector ".sidenav-user-name" to come before selector ".usa-current .sidenav-user-name"     no-descending-specificity
 34:1  ✖  Expected selector ".sidenav-user-email" to come before selector ".usa-current .sidenav-user-email"   no-descending-specificity
```

**Review by timothybest-usds — APPROVED** [review]

**Review by ClydeDroid — APPROVED** [review]
:+1:

File: frontend/src/scss/components/card.scss (hunk 1 above)
Comment (maya) [reply frontend/src/scss/components/card.scss]: Fixed in 6b1bf81

**Pull request merged**